### PR TITLE
Add Rule,Test for SLES-15-020103

### PR DESCRIPTION
**Pull request by yarunachalam**
Rule has no remediation.

#### Description:

- Add Rule,Test for SLES-15-020103

#### Rationale:

- Add SLE15 rule 'Ensure invoking users password for privilege escalation when using sudo'

### CODE REVIEW

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/oval/shared.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <ind:object object_ref="object_test_sudoers_rootpw_config" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="Ensure invoking user's password for privilege escalation when using sudo"
+  id="test_sudoers_runaspw_config" version="1">
+    <ind:object object_ref="object_test_sudoers_runaspw_config" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="object_test_sudoers_targetpw_config" version="1">
+    <ind:filepath>/etc/sudoers</ind:filepath>
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/sudo/sudoers_validate_passwd/oval/shared.xml:27]:
```suggestion
    <ind:filepath operation="pattern match">^/etc/sudoers(\.d/.*)?$</ind:filepath>
```

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/oval/shared.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <ind:pattern operation="pattern match">^Defaults !targetpw$\r?\n</ind:pattern>
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/sudo/sudoers_validate_passwd/oval/shared.xml:28]: can there be spaces at the begining? What is the reason for the \r and \n at the end?

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/oval/shared.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_object id="object_test_sudoers_rootpw_config" version="1">
+    <ind:filepath>/etc/sudoers</ind:filepath>
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/sudo/sudoers_validate_passwd/oval/shared.xml:33]:
```suggestion
    <ind:filepath operation="pattern match">^/etc/sudoers(\.d/.*)?$</ind:filepath>
```

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/oval/shared.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    <ind:pattern operation="pattern match">^Defaults !rootpw$\r?\n</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_object id="object_test_sudoers_runaspw_config" version="1">
+    <ind:filepath>/etc/sudoers</ind:filepath>
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/sudo/sudoers_validate_passwd/oval/shared.xml:39]:
```suggestion
    <ind:filepath operation="pattern match">^/etc/sudoers(\.d/.*)?$</ind:filepath>
```

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/tests/sudoers_validate_passwd_pass.sh [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# platform = SUSE Linux Enterprise 15
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/sudo/sudoers_validate_passwd/tests/sudoers_validate_passwd_pass.sh:1]: Please rename this file so that it ends in .pass.sh

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/tests/sudoers_validate_passwd_fail.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# platform = SUSE Linux Enterprise 15
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/sudo/sudoers_validate_passwd/tests/sudoers_validate_passwd_fail.sh:1]: please rename this file so that it ends in .fail.sh, othervise it won't be detected by the test suite.

File: sle15/profiles/stig.profile [hunk 7]
```diff
@@ -238,6 +238,8 @@ selections:
     - sudo_remove_nopasswd
     - sudo_restrict_privilege_elevation_to_authorized
     - sudo_require_authentication
+    - sudoers_validate_passwd
+    - susefirewall2_only_required_services
```
Comment (vojtapolasek) [inline sle15/profiles/stig.profile:242]: This line is not related to the rule, what is its purpose?

**Review by vojtapolasek — CHANGES_REQUESTED** [review]
Hello, thank you for this rule and also for tests. Please see my review comments.
Please pay attention to naming of test cases, the forat is
```
test_name.fail.sh
```

or

```
test_name.pass.sh
```

File: sle15/profiles/stig.profile (hunk 7 above)
Comment (brett060102) [reply sle15/profiles/stig.profile:242]: @yarunachalam line susefirewall2_only_required_services was removed from sle15. So, it should be removed.

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/oval/shared.xml (hunk 2 above)
Comment (yarunachalam) [reply linux_os/guide/system/software/sudo/sudoers_validate_passwd/oval/shared.xml:28]: No there cannot be spaces in the beginning and \r matches carriage return and \n matches a new line.

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/rule.yml [hunk 8]
```diff
@@ -0,0 +1,45 @@
+documentation_complete: true
+
+title: 'Ensure invoking users password for privilege escalation when using sudo'
+
+prodtype: sle15
+
+description: |-
+    The sudoers security policy requires that users authenticate themselves before they can use sudo.
+    When sudoers requires authentication, it validates the invoking user's credentials.
+severity: medium
```
[the remark this answers is not in the document]
Comment (yarunachalam) [reply linux_os/guide/system/software/sudo/sudoers_validate_passwd/rule.yml:15]: fixed

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/rule.yml [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+rationale: |-
+    If the rootpw, targetpw, or runaspw flags are defined and not disabled, by default the operating system will prompt
+    the invoking user for the "root" user password.
+
+identifiers:
+    cce@sle15: CCE-85747-4
+
+references:
+    disa@sle15: CCI-000366
```
[the remark this answers is not in the document]
Comment (yarunachalam) [reply linux_os/guide/system/software/sudo/sudoers_validate_passwd/rule.yml]: fixed

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/rule.yml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    <pre> /etc/sudoers:Defaults !targetpw
+          /etc/sudoers:Defaults !rootpw
+          /etc/sudoers:Defaults !runaspw </pre>
+
+    If no results are returned, this is a finding
+    If "Defaults !targetpw" is not defined, this is a finding.
+    If "Defaults !rootpw" is not defined, this is a finding.
+    If "Defaults !runaspw" is not defined, this is a finding.
+
+    Fix Text: Define the following in the Defaults section of the /etc/sudoers file or a configuration file in the /etc/sudoers.d/ directory:
```
[the remark this answers is not in the document]
Comment (yarunachalam) [reply linux_os/guide/system/software/sudo/sudoers_validate_passwd/rule.yml]: done

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/tests/sudoers_validate_passwd_pass.sh (hunk 5 above)
Comment (yarunachalam) [reply linux_os/guide/system/software/sudo/sudoers_validate_passwd/tests/sudoers_validate_passwd_pass.sh]: done

File: sle15/profiles/stig.profile (hunk 7 above)
Comment (yarunachalam) [reply sle15/profiles/stig.profile]: removed

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/tests/sudoers_validate_passwd_fail.sh (hunk 6 above)
Comment (yarunachalam) [reply linux_os/guide/system/software/sudo/sudoers_validate_passwd/tests/sudoers_validate_passwd_fail.sh]: done

File: linux_os/guide/system/software/sudo/sudoers_validate_passwd/rule.yml [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    disa@sle15: CCI-000366
+    nist@sle15: CM-6(b),CM-6.1(iv)
+    srg@sle15: SRG-OS-000480-GPOS-00227
+    stigid@sle15: SLES-15-020103
+
+ocil_clause: 'invoke user passwd when using sudo'
+
+ocil: |-
+    To Verify that the sudoers security policy is configured to use the invoking user's password for privilege escalation.
+
```
[the remark this answers is not in the document]
Comment (yarunachalam) [reply linux_os/guide/system/software/sudo/sudoers_validate_passwd/rule.yml]: done

**Review by vojtapolasek — APPROVED** [review]

**Comment by vojtapolasek** [comment]
Thank you for the changes. Now looks good.
